internal/resources/resources.go

```diff
@@ -0,0 +1,88 @@
+package resources
+
+import (
+	"context"
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"os"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+)
+
+type ResourceMeta struct {
+	Name      string
+	Namespace string
+}
+
+type Resource struct {
+	ResourceMeta
+	GVK schema.GroupVersionKind
+}
+
+type ResourceConfiguration struct {
+	GroupVersionKind schema.GroupVersionKind
+	ControlledList   []ResourceMeta
+}
+
+type resourceFinderConfiguration struct {
+	Resources []ResourceConfiguration
+}
+
+type IstioResourcesFinder struct {
+	ctx           context.Context
+	logger        logr.Logger
+	client        client.Client
+	configuration resourceFinderConfiguration
+}
+
+func NewIstioResourcesFinderFromConfigYaml(ctx context.Context, client client.Client, logger logr.Logger, path string) (*IstioResourcesFinder, error) {
+	configYaml, err := os.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+	var finder resourceFinderConfiguration
+	err = yaml.Unmarshal(configYaml, &finder)
+	if err != nil {
+		return nil, err
+	}
+	return &IstioResourcesFinder{
+		ctx:           ctx,
+		logger:        logger,
+		client:        client,
+		configuration: finder,
+	}, nil
+}
+
+func (i *IstioResourcesFinder) FindUserCreatedIstioResources() ([]Resource, error) {
+	var userResources []Resource
+	for _, resource := range i.configuration.Resources {
+		var u unstructured.UnstructuredList
+		u.SetGroupVersionKind(resource.GroupVersionKind)
+		err := i.client.List(i.ctx, &u)
```

Comment:
this might be not quite the best in large clusters, but since it is an uninstall do not happens often and perhaps not so much user resources at this step, and I don't have better solution, let have it like that